OpenSim/Simulation/Model/Model.cpp

```diff
@@ -489,6 +494,46 @@ void Model::createMultibodySystem()
     addToSystem(*_system);
 }
 
+
+void Model::reorderCoordinatesAccordingToSystemMobilities()
+{
+    OPENSIM_THROW_IF_FRMOBJ(!isValidSystem(), Exception,
+        "Attempting to reorder Coordinates according to an invalid System.");
+
+    auto& coordSet = updCoordinateSet();
+
+    // We have a valid MultibodySystem underlying the Coordinates
+    int nc = coordSet.getSize();
+    const SimTK::State& s = getModel().getWorkingState();
+    SimTK_ASSERT_ALWAYS(nc <= s.getNQ(),
+        "Number of Coordinates exceeds the number of mobilities in "
+        "the underlying MultibodySystem.");
+
+    auto& matter = getSystem().getMatterSubsystem();
+
+    auto coordinates = updComponentList<Coordinate>();
+
+    int cnt = 0;
+    for (auto& coord : coordinates) {
+        auto mbix = coord.getBodyIndex();
+        auto mqix = coord.getMobilizerQIndex();
+
+        int cix = matter.getMobilizedBody(mbix).getFirstUIndex(s) + mqix;
+
+        SimTK_ASSERT_ALWAYS(cix < nc, "Index exceeds the number of Coordinates "
+            "in this Model.");
+
+        // Set the coordinate in the right slot in the CoordinateSet
+        coordSet.set(cix, &coord);
+        cnt++;
+    }
+
+    SimTK_ASSERT_ALWAYS(cnt == nc,
+        "Reordered Coordinates does not correspond to the number of "
+        "Coordinates in the Model.");
```

Comment:
Unless I misunderstand this function, this check should not be necessary, given the `cix < nc` check above.